pkg/fuzz/features/https_redirects.go

```
@@ -0,0 +1,89 @@
+*/
+
+package features
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/api/networking/v1beta1"
+	"k8s.io/ingress-gce/pkg/fuzz"
+)
+
+// HTTPSRedirects
+var HTTPSRedirects = &HTTPSRedirectsFeature{}
+
+// responseCodeMap maps the GCE Response Code Names to their corresponding HTTP Response Codes
+var responseCodeMap = map[string]int{
+	"MOVED_PERMANENTLY_DEFAULT": 301,
+	"FOUND":                     302,
+	"TEMPORARY_REDIRECT":        307,
+	"PERMANENT_REDIRECT":        308,
+}
+
+// HTTPSRedirectsFeature implements the associated feature.
+type HTTPSRedirectsFeature struct {
+	fuzz.NullValidator
+
+	expectedResponseCode int
+}
+
+// Name implements fuzz.Feature.
+func (*HTTPSRedirectsFeature) Name() string {
+	return "HttpsRedirects"
+}
+
+// NewValidator implements fuzz.Feature.
+func (f *HTTPSRedirectsFeature) NewValidator() fuzz.FeatureValidator {
+	return f
+}
+
+// ConfigureAttributes implements fuzz.Feature.
+func (v *HTTPSRedirectsFeature) ConfigureAttributes(env fuzz.ValidatorEnv, ing *v1beta1.Ingress, a *fuzz.IngressValidatorAttributes) error {
+	fc, err := fuzz.FrontendConfigForIngress(ing, env)
+	if err != nil {
+		return err
+	}
+	if fc == nil || fc.Spec.RedirectToHttps == nil || !fc.Spec.RedirectToHttps.Enabled {
+		return nil
+	}
+
+	// Default
+	responseCodeName := "MOVED_PERMANENTLY_DEFAULT"
+	if fc.Spec.RedirectToHttps.ResponseCodeName != "" {
+		responseCodeName = fc.Spec.RedirectToHttps.ResponseCodeName
+	}
+
+	code, ok := responseCodeMap[responseCodeName]
+	if !ok {
+		return fmt.Errorf("error: not a valid response code name for HTTPS Redirects: %v", responseCodeName)
+	}
+	v.expectedResponseCode = code
+
+	return nil
+}
+
+// CheckResponse implements fuzz.FeatureValidator.
+func (v *HTTPSRedirectsFeature) CheckResponse(host, path string, resp *http.Response, body []byte) (fuzz.CheckResponseAction, error) {
+	if v.expectedResponseCode != 0 && resp.Request.URL.Scheme == "http" {
+		if resp.StatusCode == v.expectedResponseCode {
```

Probably better to return an explicit error? 

Otherwise it goes thru the checking if response is 200